processor/transformer/transformer.go

```diff
@@ -375,16 +354,68 @@ func (trans *HandleT) request(ctx context.Context, url string, data []Transforme
 			trans.logger.Errorf("Transformer returned : %v", string(respData))
 			respData = []byte(fmt.Sprintf("Failed to unmarshal transformer response: %s", string(respData)))
 			transformerResponses = nil
-			resp.StatusCode = 400
+			statusCode = 400
 		}
 	}
 
-	if resp.StatusCode != http.StatusOK {
+	if statusCode != http.StatusOK {
 		for i := range data {
 			transformEvent := &data[i]
-			resp := TransformerResponseT{StatusCode: resp.StatusCode, Error: string(respData), Metadata: transformEvent.Metadata}
+			resp := TransformerResponseT{StatusCode: statusCode, Error: string(respData), Metadata: transformEvent.Metadata}
 			transformerResponses = append(transformerResponses, resp)
 		}
 	}
 	return transformerResponses
 }
+
+func (trans *HandleT) doPost(ctx context.Context, rawJSON []byte, url string, tags stats.Tags) ([]byte, int) {
+	var (
+		retryCount int
+		resp       *http.Response
+		respData   []byte
+	)
+	err := backoff.RetryNotify(
+		func() error {
+			var reqErr error
+			s := time.Now()
+			trace.WithRegion(ctx, "request/post", func() {
+				resp, reqErr = trans.Client.Post(url, "application/json; charset=utf-8", bytes.NewBuffer(rawJSON))
+			})
+			trans.requestTime(tags, time.Since(s))
+
+			if reqErr == nil {
+				// If no err returned by client.Post, reading body.
+				// If reading body fails, retrying.
+				respData, reqErr = io.ReadAll(resp.Body)
+				resp.Body.Close()
+			}
+			if reqErr != nil {
+				return reqErr
+			}
+
+			return nil
+		},
+		backoff.WithMaxRetries(backoff.NewExponentialBackOff(), uint64(maxRetry)),
+		func(err error, t time.Duration) {
+			trans.logger.Errorf("JS HTTP connection error: URL: %v Error: %+v", url, err)
+			retryCount++
+		})
+	if err != nil {
+		panic(err)
```

Comment:
Why do we panic here instead of propagating the error? Assume transformation has an issue It should be handled in the same way if we get a `StatusCPDown`.

Reply:
the proposal is to retry endlessly regardless of the type of the problem?

Reply:
Agreed.
The panic wasn't introduced now, but just left in place as is to not introduce too many changes in behaviour at once.